include/galsim/SBInclinedExponentialImpl.h

```diff
@@ -0,0 +1,218 @@
+/* -*- c++ -*-
+ * Copyright (c) 2012-2016 by the GalSim developers team on GitHub
+ * https://github.com/GalSim-developers
+ *
+ * This file is part of GalSim: The modular galaxy image simulation toolkit.
+ * https://github.com/GalSim-developers/GalSim
+ *
+ * GalSim is free software: redistribution and use in source and binary forms,
+ * with or without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions, and the disclaimer given in the accompanying LICENSE
+ *    file.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions, and the disclaimer given in the documentation
+ *    and/or other materials provided with the distribution.
+ */
+
+#ifndef GalSim_SBInclinedExponentialImpl_H
+#define GalSim_SBInclinedExponentialImpl_H
+
+#include "SBProfileImpl.h"
+#include "SBInclinedExponential.h"
+#include "LRUCache.h"
+#include "Table.h"
+
+namespace galsim {
+
+    /// @brief A private class that caches the needed parameters for each inclined exponential angle 'i'
+    class InclinedExponentialInfo
+    {
```

Comment:
"As far as I can tell, the InclinedExponential calculations need to be redone for each value of h_tani_over_r" - This is correct.

As for how it might be useful, this is my imagined use case: Many simulated galaxies are needed, planned to be generated via photon shooting, with varying inclination angles (but the same h/r). If inclination angles are discretized to the nearest degree, then h_tani_over_r values will end up being reused. This means that the IFT would only need to be calculated 90 times at most (in this example), rather than once per galaxy. This could result in significant time savings.

Granted, this does depend on a few assumptions:
* The user would specifically want to do this with photon shooting
* Discretizing the inclination angle (or h_tani_over_r values) is acceptable
* There isn't a better way to do this